ApollosProject/apollos-prototype
# Adds UI Notifications Package

## DESCRIPTION

### What does this PR do, or why is it needed?

This PR moves all the `Notifications` logic into a separate package. It also restores the functionality where we update a user's PushID in Rock/OneSignal when they login. 

### What design trade-offs/decisions were made?

1. Opted **not** to refactor the `NotificationsProvider` at this time. 

### How do I test this PR?

Testing this PR is difficult. The most important thing to test is to determine if a user is being updated in OneSignal when they login in / open the app. Here is how I tested.

1. Open the iOS Simulator, log out.
2. (Open the onesignal players page)[https://app.onesignal.com/apps/b6e75a77-003f-4466-95ca-82cc5cdc407b/players]
  - Note that your simulator should be at the top of the list. Note the `external-user-id
3. Login is a _different_ user. 
  - The external user id should have changed. 

### What automated tests did you write?

Migrated existing tests

## TODO:

- [x] Closes #626

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
